src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4668,4 +4668,44 @@ public boolean isSealed() {
     }
 
     private native Class<?>[] getPermittedSubclasses0();
+
+    /*
+     * Return the class's major and minor class file version packed into an int.
+     * The high order 16 bits contain the class's minor version.  The low order
+     * 16 bits contain the class's major version.
+     *
+     * If the class is an array type then the class file version of its component
+     * type is returned.  If the class is a primitive type then the latest class
+     * file major version is returned and zero is returned for the minor version.
+     */
+    private int getClassFileVersion() {
+        Class<?> component = this;
+        if (isArray()) {
+            do {
+                component = component.getComponentType();
+            } while (component.isArray());
+        }
+        return component.getClassFileVersion0();
+    }
+
+    private native int getClassFileVersion0();
+
+    /*
+     * Return the access flags as they were in the class's bytecode, including
+     * the original setting of ACC_SUPER.
+     *
+     * If the class is an array type then the access flags of the component type is
+     * returned.  If the class is a primitive then ACC_ABSTRACT | ACC_FINAL | ACC_PUBLIC.
+     */
+    private int getClassAccessFlagsRaw() {
```

Comment:
For a "raw" method, it might be better to return the flags on the array class object itself rather than loop down to the component type.

Reply:
There's no bytecode stream for arrays.  It's created using anewarray with a dimension operand and a cp pointer to the component type.  So there are no flags for the array object.